doc/source/dev/contributor/continuous_integration.rst

```diff
@@ -0,0 +1,83 @@
+.. _continuous-integration:
+
+======================
+Continuous Integration
+======================
+
+Continuous integration is part of our development process and ensure that
+every piece of code or documentation which is contributed to SciPy is working
+and does not have unforeseen effects.
+
+.. note:: Before submitting or updating your PR, please ensure that you tested
+          locally your changes. See :ref:`pr-checklist`.
+
+Workflows
+=========
+
+We run more than 20 different workflows with different version of the
+dependencies, different architectures, etc. A PR must pass all these checks
+before it can be merged as to ensure a sustainable state of the project.
+
+Apart from the unit tests, the documentation and examples in the docstrings are
+also checked. These are common failing workflows as Sphinx and doctests have
+very strict rules. These aspects are very important as documentation and
+examples are user facing elements. Ensure that these elements are properly
+rendered.
+
+The logs can be long, but you will always find out why your did not pass a
+check. Simply click on ``Details`` to access the logs.
+
+Following is a list of all the different workflows in use. They are gouped
+by CI resources providers.
+
+GitHub Actions
+--------------
+* ``Linux Tests``: Python 3.7 and nightly Cpython 3.9
+* ``macOS Tests``: test matrix from Python 3.7 to 3.9
```

Comment:
I would make this less concrete, because otherwise it's just going to go out of date. How about:
```
* ``Linux Tests``: test suite runs for Linux (``x86_64``)
* ``macOS Tests``: test suite runs for macOS (``x86_64``)
```

"Test suite runs on GitHub Actions and other platforms cover a range of test/environment conditions: Python and NumPy versions (lowest-supported to nightly builds), 32-bit vs. 64-bit, different compilers, and more - for details, see the `.yml` configuration files".